### DISCUSSION TITLE
微软商店上架的投票

### DISCUSSION THREAD

**Discussion by bao-cn** [6get-xiaofan/DarkMode#24] [Polls]
如题

**Comment by Gaoyifei1011** [comment]
希望能以MSIX格式的安装包提交商店，而不是EXE安装包，谢谢。
如果您觉得开发Winui3的视图方面存在困难，我可以帮到您。

**Reply by Gaoyifei1011** [reply]
Winui3的任务栏托盘图标官方暂不支持，需要安装第三方开发的工具，好像叫H.TRAYICON，nuget好像就有，我本人还没试过，效果不知道怎么样。

**Reply by bao-cn** [reply]
> Winui3的任务栏托盘图标官方暂不支持，需要安装第三方开发的工具，好像叫H.TRAYICON，nuget好像就有，我本人还没试过，效果不知道怎么样。

在用WPF写了，我用的这个UI库原生支持
![image](https://user-images.githubusercontent.com/57002549/185777461-1b4d886c-d6f1-4ca8-941d-51136d4dee6e.png)

**Reply by Gaoyifei1011** [reply]
恭喜，那未来应用wpf的主界面能否现代化（win11风格），比如xaml islands

**Reply by bao-cn** [reply]
![image](https://user-images.githubusercontent.com/57002549/185778482-6d136730-c60a-40a9-8cb3-a12a071a37b7.png)

**Reply by bao-cn** [reply]
![image](https://user-images.githubusercontent.com/57002549/185778504-b98814ad-c8e1-4c86-9622-b3aa6b7f4c9a.png)

**Reply by bao-cn** [reply]
仿的微软商店界面

**Reply by Gaoyifei1011** [reply]
很美观，目前看截图好像是没有mica/acrylic，未来能否加入呢

**Reply by bao-cn** [reply]
云母材质等都是支持的，不过需要在Xaml中设置，或者通过UI库的函数调用修改

**Reply by Gaoyifei1011** [reply]
wonderful，的确很美观。那未来能否以msix格式的安装包发布到Microsoft store商店呢（毕竟appx/msix是微软商店的灵魂）。
而且那个auto dark mode在22621自动切换日夜间模式好像存在问题（据说ms内部api发生了一些变化），不知道是否会存在这些问题

**Reply by bao-cn** [reply]
![image](https://user-images.githubusercontent.com/57002549/185778695-90ea0e64-5210-46f0-886d-29b365772d18.png)
![image](https://user-images.githubusercontent.com/57002549/185778697-91a0713e-eca8-477c-b17e-3f0a9d8487ee.png)
![image](https://user-images.githubusercontent.com/57002549/185778701-54ef06ba-ad49-43ad-b8cf-5f6ba3a2ebfd.png)

**Reply by Gaoyifei1011** [reply]
原来如此。看来和winUI3存在着差别的（那个三个可以自定义设置）

**Reply by bao-cn** [reply]
> wonderful，的确很美观。那未来能否以msix格式的安装包发布到Microsoft store商店呢（毕竟appx/msix是微软商店的灵魂）。 而且那个auto dark mode在22621自动切换日夜间模式好像存在问题（据说ms内部api发生了一些变化），不知道是否会存在这些问题

如果投票支持人数多则发布多个类型，exe单文件版、exe安装程序、msix安装程序和微软商店上架的版本

**Reply by bao-cn** [reply]
> wonderful，的确很美观。那未来能否以msix格式的安装包发布到Microsoft store商店呢（毕竟appx/msix是微软商店的灵魂）。 而且那个auto dark mode在22621自动切换日夜间模式好像存在问题（据说ms内部api发生了一些变化），不知道是否会存在这些问题

确实是微软的问题，使用DarkMode的22H2用户也会存在这个问题，就连微软自家的设置中更改都有几率复现这个bug

**Reply by Gaoyifei1011** [reply]
好的，这个看您个人的时间安排和项目的关注程度了。我个人是倾向于msix的。毕竟单文件没有安装包而且如果需要开机自启（不容易被第三方软件误杀)的话msix还是很贴心的。比如那个snipaste软件在商店发布的就是msix。倒也不排除其他人喜欢其他打包方式

**Reply by Gaoyifei1011** [reply]
是的，不知道您个人在测试的时候是否遇到了这个问题？系统版本是不是22621？

**Reply by Gaoyifei1011** [reply]
此外，上架商店的软件。如果更新频率较快，exe独立包很不友好，不能自动更新。商店的msix方案可以很好的省去开发者的这些考虑😂😂

**Reply by bao-cn** [reply]
我是21H2，没有此问题，但经过虚拟机测试，确实有此问题，已经确定问题是微软的Bug。感谢建议
